cli/cache/module_info.rs

```diff
@@ -134,20 +132,46 @@ impl ModuleInfoCache {
     source_hash: &ModuleInfoCacheSourceHash,
     module_info: &ModuleInfo,
   ) -> Result<(), AnyError> {
-    let sql = "
-      INSERT OR REPLACE INTO
-        moduleinfocache (specifier, media_type, source_hash, module_info)
-      VALUES
-        (?1, ?2, ?3, ?4)";
-    self.conn.execute(
-      sql,
-      params![
-        specifier.as_str(),
-        serialize_media_type(media_type),
-        source_hash.as_str(),
-        &serde_json::to_string(&module_info)?,
-      ],
-    )?;
+    let data = (
+      specifier.to_string(),
+      serialize_media_type(media_type),
+      source_hash.as_str().to_string(),
+      serde_json::to_string(&module_info).unwrap(),
+    );
+
+    let mut sender = self.sender.lock();
+    if let Some(sender) = &mut *sender {
+      let _ = sender.send(data);
+    } else {
+      let (tx, mut rx) = tokio::sync::mpsc::unbounded_channel();
+      let conn = self.conn.clone();
+      let _ = deno_core::unsync::spawn(async move {
+        while let Some((specifier, media_type, source_hash, module_info)) =
+          rx.recv().await
+        {
+          let conn = conn.clone();
+          deno_core::unsync::spawn_blocking(move || {
+            let sql = "
+            INSERT OR REPLACE INTO
+              moduleinfocache (specifier, media_type, source_hash, module_info)
+            VALUES
+              (?1, ?2, ?3, ?4)";
+            let result = conn.execute(
+              sql,
+              params![specifier, media_type, source_hash, module_info],
+            );
+            if let Err(err) = result {
+              log::debug!("Error saving module cache info. {:#}", err);
+            }
+          })
+          .await
+          .unwrap();
+        }
+      });
+      tx.send(data).unwrap();
+      *sender = Some(tx);
+    }
```

Comment:
This is not a good solution though because it will cause this information to not be saved to the cache if the application shuts down fast enough (actually, it's not entirely clear that the save occurs in the benchmark I posted). Probably we need some variant on this solution (ex. in the case of code cache making it async might work well)

Reply:
Confirmed actually that this does not save the cache in time before shutdown in the benchmark.